app/src/main/java/com/example/mafkir/MainActivity.kt

```diff
@@ -1,12 +1,68 @@
 package com.example.mafkir
 
-import androidx.appcompat.app.AppCompatActivity
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import android.content.Intent
+import android.os.Build
 import android.os.Bundle
+import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.NotificationCompat
+import androidx.core.app.NotificationManagerCompat
+import com.example.mafkir.notifications.NotificationRegistry
+import com.example.mafkir.notifications.NotificationsService
+import com.google.common.eventbus.EventBus
+import dagger.android.AndroidInjection
+import javax.inject.Inject
+import javax.inject.Named
 
 class MainActivity : AppCompatActivity() {
 
+    @Inject @field:Named("Notifications")
+    lateinit var mEventBus: EventBus;
+
     override fun onCreate(savedInstanceState: Bundle?) {
+        AndroidInjection.inject(this);
         super.onCreate(savedInstanceState)
+        mEventBus.register(NotificationRegistry());
+        createNotificationChannel();
+        startService(Intent(this, NotificationsService::class.java))
         setContentView(R.layout.activity_main)
+        val button = findViewById<Button>(R.id.testNotification)
+        button.setOnClickListener{
+            onClickNotify()
+        }
+    }
+
+    private fun createNotificationChannel() {
```

Comment:
maybe rename to `registerNotificationChannel` and if u feel lik it move to a different class (this class will have a lot to do with gui too in the future)